sites/website/src/docs/design-systems/creating-a-component-library.md

```diff
@@ -0,0 +1,117 @@
+- Custom element tag names are easily configured and disambiguated.
+- Custom elements are explicitly registered to avoid tree-shaking.
+- templates, styles, default slotted content, and arbitrary component configurations are all configurable by application authors.
+- Shadow DOM options are easily configurable.
+All these configuration capabilities are designed to give application authors the flexibility necessary to align library components to application or business requirements.
+This section walks through the process of defining a component from a library author’s perspective, detailing component, template, and style creation that supports the configuration capabilities described above.
+
+## Defining components
+Defining a component for a library is easy; simply extend the `FoundationElement` base class from `@microsoft/fast-foundation`. `FoundationElement` is a light extension of `FASTElement` that supports element configuration during component registration.
+
+```ts
+import { FoundationElement } from “@microsoft/fast-foundation”;
+
+export class MyElement extends FoundationElement {
+	/* component implementation */
+}
+```
+
+### Define Templates and Styles 
+There are two ways to define templates and styles. 
+Just like `FASTElement`, a component that extends `FoundationElement` can be used with the [`html`](/docs/fast-element/declaring-templates) and [`css`](/docs/fast-element/leveraging-css) tagged template literals.
+
+```ts
+import { html, css  } from "@microsoft/fast-element";
+
+const myElementStyles = css`/* ... */`;
+const myElementTemplate = html`<!-- ... -->`;
+```
+
+A `FoundationElement` also supports a lazily-defined template and style type. This lazy option is a function invoked with an [ElementDefinitionContext](/docs/api/fast-foundation.elementdefinitioncontext) and [FoundationElementDefinition](/docs/api/fast-foundation.foundationelementdefinition) options, providing opportunity to adjust the template or styles based on how the element or other elements are defined. Lazy template and style definitions provide incredible flexibility, and is what can allow application authors to re-name component tag names, override default slotted content, and communicate any component-specific configuration options:
+
+```ts
+import { html, css  } from "@microsoft/fast-element";
+import { ElementDefinitionContext, FoundationElementDefinition } from "@microsoft/fast-foundation";
+
+const myElementStyles = (
+context: ElementDefinitionContext,
+definition: FoundationElementDefinition)
+ => css`/* ... */`;
+const myElementTemplate = (
+context: ElementDefinitionContext,
+definition: FoundationElementDefinition
+) => html`<!-- ... -->`;
+```
+
+Let’s take a closer look at what these two arguments give us.
+
+#### What is the `ElementDefinitionContext`?
+// TODO Rob - I'm not sure how to explain this argument well or what all of it's capabilities are
```

Comment:
@EisenbergEffect I wasn't exactly sure how to describe the ElementDefinitionContext - can you take a stab at it?